.cloudbuild/ci/integration-tests.yaml

```diff
@@ -1,29 +1,69 @@
+timeout: 20m
+options:
+  machineType: E2_HIGHCPU_32
+  
+  volumes:
+    # A place to put shared state between build steps so that we don't pollute
+    # the workspace
+    - name: state
+      path: '/state'
+
+  env: 
+    # Path to where the Go module cache will live, where we can control it
+    - GOMODCACHE=/state/gomodcacheci
+
+    # Path to a marker file. If this file exists test steps should be skipped.
+    - SKIP_TESTS=/state/skip
+
 steps:
+  # GCB does a shallow checkout for a build, but if we want to check our changes
+  # against other branches we'll need to fetch the repo history.
+  - name: gcr.io/cloud-builders/git
+    id: fetch-history
+    args: ['fetch', '--unshallow']
+
+  # Check if the PR only has docs changes. If so we will set a marker file 
+  # meaning that we can skip the other tests in this target
+  - name: golang:1.17.5
+    id: check-tests
+    dir: /workspace/.cloudbuild/scripts
+    entrypoint: go
+    args: [run, ./can-skip, -w=/workspace, -s=$$SKIP_TESTS, -t=$_BASE_BRANCH, -c=HEAD]
+
   # Run non-root integration tests. The root user is selected by docker image.
   - name: us-docker.pkg.dev/ci-account/teleport/buildbox-root:v0.1.0
     id: root-integration-test
-    args: ['make', 'rdpclient', 'integration-root']
+    entrypoint: bash
+    args: 
+      - '-c'
+      - |
+        if [ ! -f $$SKIP_TESTS ]; then 
+          make rdpclient integration-root
+        fi
```

Comment:
I think this is fine for now to unblock docs writers, but I think we should just execute a program like your `can-skip` as a wrapper checks if its a docs PR and running unit and integration tests can be skipped. If it not, it can just `os.exec` `make tests`.

This has several benefits. It makes each run stateless and you don't need to persist a `/state` directory.

This approach has another benefit, you can get rid of `(make run-etcd &); sleep 1; make integration` and instead launch etcd, wait for etcd to be healthy https://etcd.io/docs/v3.4/op-guide/monitoring/#health-check then run tests.

Basically let's follow a similar approach to what we are doing with the workflow automation bot, reduce as much logic in YAML, push it all to a thin Go program.